src/cluster/kv/etcd/store_test.go

```diff
@@ -125,6 +126,27 @@ func TestNoCache(t *testing.T) {
 	require.Equal(t, 0, len(store.(*client).cacheUpdatedCh))
 }
 
+func TestCacheDirCreation(t *testing.T) {
+	ec, opts, closeFn := testStore(t)
+
+	tdir, err := ioutil.TempDir("", "m3tests")
+	defer os.RemoveAll(tdir)
+	require.NoError(t, err)
+
+	cdir := path.Join(tdir, randSeq(10))
```

Comment:
not sure why you need the randSeq when you're already using a new random test directory

Reply:
I think when calling `ioutil.TempDir`, it will already create a random directory. So I'm using `TempDir` and then choose a random dir name `cdir` be created by `NewStore` into `TempDir`. Then I can test for the existence of `cdir`.

Reply:
nit: this feels like an overkill, a const name would do here.

Reply:
Ok I can just use a const. I was thinking that there might be a case where this test could be run in parallel on the same machine, which might lead to multiple tests trying to create the same directory. But maybe I was overthinking :)